src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamWriteAsyncMemoryOverloadsTests.cs

```diff
@@ -0,0 +1,711 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferStreamWriteAsyncMemoryOverloadsTest : PreferStreamAsyncMemoryOverloadsTestBase
+    {
+        #region C# - No diagnostic - Analyzer
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_Write()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    void M()
+    {
+        byte[] buffer = { 0xBA, 0x5E, 0xBA, 0x11, 0xF0, 0x07, 0xBA, 0x11 };
+        using (FileStream fs = new FileStream(""path.txt"", FileMode.Create))
+        {
+            fs.Write(buffer, 0, buffer.Length);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_WriteAsync_ByteMemory()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = new FileStream(""path.txt"", FileMode.Create))
+        {
+            byte[] buffer = { 0xBA, 0x5E, 0xBA, 0x11, 0xF0, 0x07, 0xBA, 0x11 };
+            Memory<byte> memory = new Memory<byte>(buffer);
+            await fs.WriteAsync(memory, new CancellationToken()).ConfigureAwait(false);
```

Comment:
Similar to the read tests; the write tests DO want to report a diagnostic when it's await/ConfigureAwait.

Reply:
This specific case should not give a diagnostic because the WriteAsync method is the preferred overload, not the undesired one.